File: monorepo_csharp/packages/Shared/Chain.cs
namespace Mono.Shared.Chain;

// T1.origin — definition lives here.
public static class Origin
{
    public const string ORIGIN_VALUE = "T1.origin";         // T1.origin.def
}

// T1.middle — re-exposes origin's value as its own constant.
public static class Middle
{
    public const string MIDDLE_VALUE = Origin.ORIGIN_VALUE; // T1.middle.reexport
}

// T1.deep — final hop, renamed.
public static class Deep
{
    public const string VALUE_ALIAS = Middle.MIDDLE_VALUE;  // T1.deep.reexport
}

File: monorepo_kotlin/packages/shared/src/main/kotlin/mono/shared/chain/Chain.kt
package mono.shared.chain

// T1.origin — defined here.
object Origin {
    const val ORIGIN_VALUE: String = "T1.origin"                 // T1.origin.def
}

// T1.middle — re-exposes Origin via its own constant.
object Middle {
    const val MIDDLE_VALUE: String = Origin.ORIGIN_VALUE         // T1.middle.reexport
}

// T1.deep — renamed alias at the third level.
object Deep {
    const val VALUE_ALIAS: String = Middle.MIDDLE_VALUE          // T1.deep.reexport
}

// Extra: `typealias` as another re-export shape.
typealias AliasedOrigin = Origin
